apps/src/javalab/Javalab.js

```diff
@@ -269,6 +270,10 @@ Javalab.prototype.init = function(config) {
   getStore().dispatch(setIsStartMode(this.isStartMode));
   getStore().dispatch(setLevelName(this.level.name));
 
+  // For javalab, we don't use pageConstants.isReadOnlyWorkspace because
+  // the readOnly state can change when a code review is opened or closed
+  getStore().dispatch(setIsReadOnlyWorkspace(!!config.readonlyWorkspace));
```

Comment:
There is going to need to be an aspect of it that's javalab-specific because there is a javalab editor. I'm not sure we have time to allow every editor to be able to have a variable readonly setting. I'll check in studioApp though because it's possible there are settings I missed that need to be variable

Reply:
Ok, maybe we can have someone from CSA who's more familiar than I am with javalab take a look and see if they're comfortable with Javalab being different in this way?

(@breville, tagging you as an FYI as you're thinking about LabLab.)

Reply:
I'm ok with this--Java Lab has a fair amount of unique aspects, and since we don't call studioApp.init I don't think there's a risk here of things getting confused. We also don't do anything with pairing.